Fixes the redeclaration of ttyname which was preventing
enabling clang fortify.

The error was;

main.c:858:15: error: redeclaration of 'ttyname' must have the 'overloadable' attribute
    char *p, *ttyname();
              ^
/build/samus/usr/include/unistd.h:784:14: note: previous overload of function is here
extern char *ttyname (int __fd) __THROW __CLANG_NO_MANGLE (ttyname);

https://github.com/krb5/krb5/pull/568

Patch by Zentaro Kavanagh <zentaro@google.com>

--- clients/ksu/main.c
+++ clients/ksu/main.c
@@ -855,7 +855,7 @@
 
 static char * ontty()
 {
-    char *p, *ttyname();
+    char *p;
     static char buf[MAXPATHLEN + 5];
     int result;
 
